var/spack/repos/builtin/packages/hdf5-vol-async/package.py

```diff
@@ -24,15 +26,17 @@ class Hdf5VolAsync(CMakePackage):
     version("1.0", tag="v1.0")
 
     depends_on("mpi")
-    depends_on("argobots@main")
+    depends_on("argobots@1.1:")
     depends_on("hdf5 +mpi +threadsafe")
     depends_on("hdf5@1.13.0:1.13.2", when="@:1.3")
     depends_on("hdf5@1.13.3:", when="@1.4:")
 
+    # Enforce that MPI_THREAD_MULTIPLE is available.
+    depends_on("openmpi +thread_multiple", when="^openmpi")
+    depends_on("mvapich2 threads=multiple", when="^mvapich2")
+
     def setup_run_environment(self, env):
-        env.set("HDF5_PLUGIN_PATH", self.spec.prefix.lib)
-        vol_connector = "async under_vol=0;under_info=[]"
-        env.set("HDF5_VOL_CONNECTOR", vol_connector)
```

Comment:
Maybe I missed this, but why is adding the warning in the docstring better than removing setting `HDF5_VOL_CONNECTOR` when setting up the run environment? the docstring

Reply:
`HDF5_VOL_CONNECTOR` is an application/runtime choice, setting it for the runtime environment is too heavy handed. It is possible to have multiple VOLs activated in the same environment or package as well as potentially overwriting user configurations silently.

Reply:
Hmm.  So having a "default" setting is too heavy handed?  

What about stand-alone testing?  If/when stand-alone tests are added to the package someone will have to be sure to set the environment accordingly.  Setting up the runs includes Spack automatically calling `setup_run_environment`.

Reply:
I could see the benefit for the stand-alone testing to have a helper function that provided the default `HDF5_VOL_CONNECTOR` string.

Not sure if this is idiomatic Spack or not..

```python
env.set("HDF5_VOL_CONNECTOR", self.spec["hdf5_vol_async"].package.default_vol_connector)
```

But even in stand-alone testing, the vol connector is a user configuration, so it should be intentionally set by the application using it so it is clear that that test is using the Async VOL and not the Cache VOL (which would also be set when this is loaded, not sure on ordering) or no VOL for one test and with a VOL for a different test.